services/flathub/flathub.service.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { InvalidResponse, NotFound } = require('..')
+const { renderVersionBadge } = require('../version')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  currentReleaseVersion: Joi.string().required(),
+}).required()
+
+module.exports = class Flathub extends BaseJsonService {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'flathub/v',
+      pattern: ':packageName',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Flathub',
+        namedParams: {
+          packageName: 'org.mozilla.firefox',
+        },
+        staticPreview: renderVersionBadge({ version: '78.0.2' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'flathub' }
+  }
+
+  async handle({ packageName }) {
+    try {
+      const data = await this._requestJson({
+        schema,
+        url: `https://flathub.org/api/v1/apps/${encodeURIComponent(
+          packageName
+        )}`,
+      })
+      return renderVersionBadge({ version: data.currentReleaseVersion })
+    } catch (error) {
+      if (error instanceof InvalidResponse) {
+        throw new NotFound({
```

Comment:
After a bit experementing, I can see why we're doing this: The 'not found' response from FlatHub is a 200 OK with an empty body so we're re-throwing the parse error as a `NotFound`, but its a bit non-standard. When we have to deal with an odd API like this, we usually add a short comment to explain what we're doing e.g:
https://github.com/badges/shields/blob/478d14300c4203a134ddf8cbaf5bc1ee8150e5a3/services/cdnjs/cdnjs.service.js#L52-L56
https://github.com/badges/shields/blob/478d14300c4203a134ddf8cbaf5bc1ee8150e5a3/services/itunes/itunes.service.js#L51-L56
Can we do the same here.